tests/lib/rules/no-extra-parens.js

```diff
@@ -314,6 +314,10 @@ ruleTester.run("no-extra-parens", rule, {
         { code: "foo + (bar + baz)", options: ["all", { nestedBinaryExpressions: false }] },
         { code: "foo && (bar && baz)", options: ["all", { nestedBinaryExpressions: false }] },
 
+        // https://github.com/eslint/eslint/issues/9019
+        { code: "(async function() {});" },
+        { code: "(async function () { }());" },
+
```

Comment:
Can you add a test case for the following code?

```js
(async)
```

This should be invalid, since the parentheses are unnecessary. (`async` is used as an identifier here.)